src/cavendish_particle_tracks/_set_fiducial_dialog.py

```diff
@@ -0,0 +1,536 @@
+
+    def _on_change_cmb_set_ref_plane(self) -> None:
+        # check how original code works and make sure I've copied the correct functionality.
+        if self.cmb_set_ref_plane.currentIndex == 0:
+            self.label_stereoshift.setText(
+                "Stereo shift (shift_p/shift_f = depth_p/depth_f)"
+            )
+        elif self.cmb_set_ref_plane.currentIndex == 1:
+            self.label_stereoshift.setText(
+                "Stereo shift (shift_p/shift_f = 1 - depth_p/depth_f)"
+            )
+
+        self.layer_points.refresh()
+
+    # def _on_change_cmb_fiducial etc etc
+    # add the label to the point in the view
+    # replaces the old on_click_add button
+
+    def _setup_points_layer(self) -> napari.layers.Points:
+        # TODO: Is there a better way of typehinting without importing the entirety of Napari>
+        # I mean, I assume this import doesn't actually use any extra memory since Napari is imported anyway
+        # whilst this plugin is running. But I don't know enough about how Python handles imports / memory usage to be sure.
+        # add the points
+        points = np.array(  # TODO points is a duplicate of the fiducial array, why bother keeping both?
+            [
+                [100, 100],  # Front Fiducial 1 View 1
+                [101, 400],  # Front Fiducial 1 View 2
+                [200, 100],  # Front Fiducial 2 View 1
+                [200, 102],  # Front Fiducial 2 View 2
+                [202, 399],  # Point View 1
+                [300, 103],  # Point View 2
+                [299, 401],  # Back Fiducial 1 View 1
+                [400, 402],  # Back Fiducial 1 View 2
+                [500, 400],  # Back Fiducial 2 View 1
+                [500, 402],  # BackFiducial 2 View 2
+                # then expand this to include the magnification points and redo the array
+            ]
+        )
+        # TODO fix all this here and check how it's actually working
+        # TODO implment the layer attributes stuff see if i can integrate it with the existing napari stuff
+        labels = ["Reference (Front) view1", "Reference (Front) view2"]
+        for i in self.points:
+            # What I've done here is a big no-no, as the list of fiducials here
+            # carries the same name as the points used to instantiate the layer.
+            # (ie. self.points = fiducials, points = the array above as the scope is limited to this function)
+            # This is kinda why I want to change this, but leaving it as functional but odd till
+            # we figure out how to get the points to instantiate in the middle of the viewport or
+            # decide otherwise.
+            for j in i:
+                labels += [j.name]
+
+        colors = [
+            "green",
+            "green",
+            "yellow",
+            "yellow",
+            "blue",
+            "blue",
+            "red",
+            "red",
+            "orange",
+            "orange",
+        ]
```

Comment:
I would keep the colour to indicate the view and use different shapes for the different type of point, i.e. Front fiducial, Back fiducial, Point of interest.